src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -4291,11 +4296,11 @@ void QgsPostgresSharedData::setFeaturesCounted( long count )
 }
 
 
-QgsFeatureId QgsPostgresSharedData::lookupFid( const QVariant &v )
+QgsFeatureId QgsPostgresSharedData::lookupFid( const QVariantList& v )
 {
   QMutexLocker locker( &mMutex );
 
-  QMap<QVariant, QgsFeatureId>::const_iterator it = mKeyToFid.constFind( v );
+  QMap<QVariantList, QgsFeatureId>::const_iterator it = mKeyToFid.constFind( v );
```

Comment:
Yeah, I've been concerned about the performance impact of forcing everything through this map too.